### ISSUE TITLE
读取batch时报错，ctrl+c无法暂停。

### ISSUE THREAD

**Issue by Chucy2020** [PaddlePaddle/Paddle#43631]
### bug描述 Describe the Bug

这是代码
![UPI TZEX~{M(E$1H I{@ZHF](https://user-images.githubusercontent.com/73978694/174265178-47cbb9fa-5f95-444d-af35-893c7b0bf6c6.png)
下面是报错
![VW6@K7QW KS{3_97)428 VW](https://user-images.githubusercontent.com/73978694/174265261-da15612c-7358-4083-92bb-8c78091d1888.png)

### 其他补充信息 Additional Supplementary Information

paddle2.2

**Comment by linjieccc** [comment]
您好，能否提供下最小可复现代码

**Comment by Chucy2020** [comment]
> 您好，能否提供下最小可复现代码

![image](https://user-images.githubusercontent.com/73978694/174275855-0fe9b77a-a688-499b-a618-c93be11b2fc6.png)
![image](https://user-images.githubusercontent.com/73978694/174275927-2cc0cb2f-e687-4f55-a5e8-686ead133c12.png)
shang'm上面是我运行的代码
![A~IEJL208B QW8RI7V`%RYE](https://user-images.githubusercontent.com/73978694/174276005-e171e56b-2561-4790-94c4-223734a07732.jpg)
数据格式如下，下面是一个yang'li样例，上面是n个样本。

**Comment by Chucy2020** [comment]
> > 您好，能否提供下最小可复现代码
> 
> ![image](https://user-images.githubusercontent.com/73978694/174275855-0fe9b77a-a688-499b-a618-c93be11b2fc6.png) ![image](https://user-images.githubusercontent.com/73978694/174275927-2cc0cb2f-e687-4f55-a5e8-686ead133c12.png) shang'm上面是我运行的代码 ![A~IEJL208B QW8RI7V`%RYE](https://user-images.githubusercontent.com/73978694/174276005-e171e56b-2561-4790-94c4-223734a07732.jpg) 数据格式如下，下面是一个yang'li样例，上面是n个样本。

![image](https://user-images.githubusercontent.com/73978694/174276279-0c8a9b4b-5cd1-40cf-83e6-db94003495cf.png)
这是一个样本的样例。

**Comment by yt605155624** [comment]
请把代码和数据粘贴出来我们才好帮你复现，否则我们还要手动敲代码，很浪费时间

**Comment by Chucy2020** [comment]
> 

好的，不好意思哈，现在我将代码复制下来了。如下。麻烦大佬们帮忙解答疑惑，感谢。
import paddle
from paddlenlp.data import Stack, Dict, Pad
from paddlenlp.datasets import MapDataset

batch_size = 2
data_list = []
data_path="data.txt"
with open(data_path,encoding="utf-8") as f:
    for line in f.readlines():
        one_exapme = line.strip()
        data_list.append(eval(one_exapme))

data_list = MapDataset(data_list)

 for idx in range(1):
     print(data_list[idx]['input_ids'])
     print(data_list[idx]['input_mask'])
     print(data_list[idx]['segment_ids'])
     print(data_list[idx]['start_positions'])
     print(data_list[idx]['end_positions'])

train_batch_sampler = paddle.io.DistributedBatchSampler(
        data_list, batch_size=batch_size, shuffle=True)

train_batchify_fn = lambda samples, fn=Dict({
    "input_ids": Pad(axis=0, pad_val=0),
    "input_mask": Pad(axis=0, pad_val=0),
    "segment_ids": Pad(axis=0, pad_val=0),
    "start_positions": Stack(dtype="int64"),
    "end_positions": Stack(dtype="int64")
}): fn(samples)

train_data_loader = paddle.io.DataLoader(
    dataset=data_list,
    batch_sampler=train_batch_sampler,
    collate_fn=train_batchify_fn,
    return_list=True)

for step, batch in enumerate(train_data_loader, start=1):
    input_ids, input_mask, segment_ids, start_positions, end_positions = batch
    print(input_ids)
    break



#数据demo如下
[data.txt](https://github.com/PaddlePaddle/Paddle/files/8927770/dev384.data.txt)

数据demo
{'input_ids':[1, 1034, 1189, 734, 2003, 241, 284, 131, 553],'input_mask':[1, 1, 1, 1, 1, 1, 1, 1, 1],'segment_ids':[0, 0, 0, 1, 1, 1, 1, 1, 1],'start_positions':5,'end_positions':7}
{'input_ids':[1, 1034, 1189, 734, 2003, 241, 284,],'input_mask':[1, 1, 1, 1, 1, 1, 1],'segment_ids':[0, 0, 0, 1, 1, 1, 1],'start_positions':2,'end_positions':4}
{'input_ids':[1, 1034, 1189, 734, 2003, 241, 284, 131],'input_mask':[1, 1, 1, 1, 1, 1, 1, 1],'segment_ids':[0, 0, 0, 1, 1, 1, 1, 1],'start_positions':4,'end_positions':6}
{'input_ids':[1, 1034, 1189, 734, 2003, 241],'input_mask':[1, 1, 1, 1, 1, 1],'segment_ids':[0, 0, 0, 1, 1, 1],'start_positions':2,'end_positions':3}

**Comment by Chucy2020** [comment]
> 请把代码和数据粘贴出来我们才好帮你复现，否则我们还要手动敲代码，很浪费时间

import paddle
from paddlenlp.data import Stack, Dict, Pad
from paddlenlp.datasets import MapDataset

batch_size = 16
data_list = []
data_path="dev384.data"
with open(data_path,encoding="utf-8") as f:
    for line in f.readlines():
        one_exapme = line.strip()
        data_list.append(eval(one_exapme))

data_list = MapDataset(data_list)

for idx in range(2):
    print(data_list[idx]['input_ids'])
    print(data_list[idx]['input_mask'])
    print(data_list[idx]['segment_ids'])
    print(data_list[idx]['start_positions'])
    print(data_list[idx]['end_positions'])

train_batch_sampler = paddle.io.DistributedBatchSampler(
        data_list, batch_size=batch_size, shuffle=True)

train_batchify_fn = lambda samples, fn=Dict({
    "input_ids": Pad(axis=0, pad_val=0),
    "input_mask": Pad(axis=0, pad_val=0),
    "segment_ids": Pad(axis=0, pad_val=0),
    "start_positions": Stack(dtype="int64"),
    "end_positions": Stack(dtype="int64")
}): fn(samples)

###构造DataLoader
train_data_loader = paddle.io.DataLoader(
    dataset=data_list,
    batch_sampler=train_batch_sampler,
    collate_fn=train_batchify_fn,
    return_list=True)

for step, batch in enumerate(train_data_loader, start=1):

    input_ids, segment_ids, start_positions, end_positions = batch
    print(input_ids)
    break

**Comment by Chucy2020** [comment]
> 您好，能否提供下最小可复现代码

麻烦您抽时间看下这个是什么问题？感谢，每次运行for step, batch in enumerate(train_data_loader):就会陷入死循环，ctrl+c无法强制暂停。

**Comment by Chucy2020** [comment]
> 请把代码和数据粘贴出来我们才好帮你复现，否则我们还要手动敲代码，很浪费时间

麻烦您抽时间看下这个是什么问题？感谢，每次运行for step, batch in enumerate(train_data_loader):就会陷入死循环，ctrl+c无法强制暂停。

**Comment by linjieccc** [comment]
请问该情况是可以稳定复现的么，我这边用您提供的数据demo测试了下是可以正常运行的，通过ctrl-C可以立刻终止。

**Comment by Chucy2020** [comment]
> 请问该情况是可以稳定复现的么，我这边用您提供的数据demo测试了下是可以正常运行的，通过ctrl-C可以立刻终止。

![image](https://user-images.githubusercontent.com/73978694/174544145-d629e411-19b5-4a38-8636-568c2083d306.png)

感谢您抽时间解答我的问题，我这里还是那个问题，当运行那个for step, batch in enumerate(train_data_loader):时，无法执行之后的程序，而且一直无法强制暂停，直到一定时间后程序被调试机强制killed。不知道这是怎么回事。

**Comment by Chucy2020** [comment]
> 请问该情况是可以稳定复现的么，我这边用您提供的数据demo测试了下是可以正常运行的，通过ctrl-C可以立刻终止。

您好，感谢大佬们抽出宝贵时间为我们解答，我又测试了这个代码，现在可以了，但是前面为什么会一直无法强制停止的问题还是不知道，还有上面时另一个类似代码运行出现无法强制停止的例子，可能是我自己的原因触发了什么吧。
